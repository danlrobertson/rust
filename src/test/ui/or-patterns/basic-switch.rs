// run-pass
#![feature(or_patterns)]
//~^ WARN the feature `or_patterns` is incomplete and may cause the compiler to crash

#[derive(Debug)]
enum Test {
    Foo,
    Bar,
    Baz,
    Qux
}

#[derive(Debug)]
enum Wrapper {
    Filled(Test),
    Empty,
}

fn test_wrapper(x: Wrapper) -> bool {
    match x {
        // most simple case
        Wrapper::Filled(Test::Bar | Test::Qux) => true,
        // wild case
        Wrapper::Filled(_) => false,
        // empty case
        Wrapper::Empty => false,
    }
}

fn main() {
    assert!(!test_wrapper(Wrapper::Filled(Test::Foo)));
    assert!(test_wrapper(Wrapper::Filled(Test::Bar)));
    assert!(!test_wrapper(Wrapper::Filled(Test::Baz)));
    assert!(test_wrapper(Wrapper::Filled(Test::Qux)));
    assert!(!test_wrapper(Wrapper::Empty))
}
