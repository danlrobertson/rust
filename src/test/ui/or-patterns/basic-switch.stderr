warning: the feature `or_patterns` is incomplete and may cause the compiler to crash
  --> $DIR/basic-switch.rs:2:12
   |
LL | #![feature(or_patterns)]
   |            ^^^^^^^^^^^

